test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatWhyTests.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.CommandLine.XPlat;
+using NuGet.Packaging;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.XPlat.FuncTest
+{
+    [Collection("NuGet XPlat Test Collection")]
+    public class XPlatWhyTests
+    {
+        private static readonly string ProjectName = "Test.Project.DotnetNugetWhy";
+        private static MSBuildAPIUtility MsBuild => new MSBuildAPIUtility(new TestCommandOutputLogger());
+
+        [Fact]
+        public async void WhyCommand_BasicFunctionality_Succeeds()
```

Comment:
I prefer names that are more explicit. Just a suggestion:
```suggestion
        public async void WhyCommand_WhenProjectHasDependencyGraph_Succeeds()
```

Reply:
I agree and I'd also suggest adding a more descriptive suffix explaining what it means for this test to succeed.

Example:
Let's say we have a test for an indexOf method and we call our test IndexOf_WhenCharacterExistsInString_Succeeds, we can instead name it something more descriptive like what it means for this test to succeed. Something like this:
IndexOf_WhenCharacterExistsInString_ReturnsFirstFoundPosition

Reply:
Thanks! I've taken another run at naming these tests. Please let me know if you still think it could do be better, and I'd be happy to make further changes!